tests/conftest.py

```diff
@@ -0,0 +1,139 @@
+import aiomcache
+import aioredis
+import asyncio
+import gc
+import pytest
+import time
+import uuid
+from docker import Client as DockerClient
+
+
+@pytest.fixture(scope='session')
+def loop(request):
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(None)
+
+    yield loop
+
+    if not loop._closed:
+        loop.call_soon(loop.stop)
+        loop.run_forever()
+        loop.close()
+    gc.collect()
+    asyncio.set_event_loop(None)
+
+
+@pytest.fixture(scope='session')
+def session_id():
+    '''Unique session identifier, random string.'''
+    return str(uuid.uuid4())
+
+
+@pytest.fixture(scope='session')
+def docker():
+    return DockerClient(version='auto')
+
+
+def pytest_addoption(parser):
+    parser.addoption("--no-pull", action="store_true", default=False,
+                     help="Don't perform docker images pulling")
+
+
+@pytest.fixture(scope='session')
+def redis_server(docker, session_id, loop, request):
+    if not request.config.option.no_pull:
+        docker.pull('redis:{}'.format('latest'))
+    container = docker.create_container(
+        image='redis:{}'.format('latest'),
+        name='redis-test-server-{}-{}'.format('latest', session_id),
+        ports=[6379],
+        detach=True,
+    )
+    docker.start(container=container['Id'])
+    inspection = docker.inspect_container(container['Id'])
+    host = inspection['NetworkSettings']['IPAddress']
+    delay = 0.001
+    for i in range(100):
+        try:
+            conn = loop.run_until_complete(
+                aioredis.create_connection((host, 6379))
```

Comment:
in order to fix 
```
RuntimeError: There is no current event loop in thread 'MainThread'.
```
you need to pass loop explicitly here

Reply:
To avoid multiple pushes, you mean pass it to `run_until_complete` as a kwarg right?

Reply:
(retracted run_until_complete doesn't take loop kwarg :s so I guess in create_connection)